gen/generator.go

```diff
@@ -1660,14 +1663,38 @@ Precendence order is:
  2. global level from properties.yaml
 */
 func requiredProperty(propertyName, classPkgName string, definitions Definitions) bool {
+	// precedenceList := []string{classPkgName, "global"}
+	// for _, precedence := range precedenceList {
+	// 	if classDetails, ok := definitions.Properties[precedence]; ok {
+	// 		for key, value := range classDetails.(map[interface{}]interface{}) {
+	// 			if key.(string) == "resource_required" {
+	// 				for _, v := range value.([]interface{}) {
+	// 					if v.(string) == propertyName {
+	// 						return true
+	// 					}
+	// 				}
+	// 			}
+	// 		}
+	// 	}
+	// }
+	// return false
 	precedenceList := []string{classPkgName, "global"}
 	for _, precedence := range precedenceList {
 		if classDetails, ok := definitions.Properties[precedence]; ok {
 			for key, value := range classDetails.(map[interface{}]interface{}) {
 				if key.(string) == "resource_required" {
-					for _, v := range value.([]interface{}) {
-						if v.(string) == propertyName {
-							return true
+					v1, ok1 := value.([]interface{})
+					if !ok1 {
+						for k1, _ := range value.(map[interface{}]interface{}) {
+							if k1.(string) == propertyName {
+								return true
+							}
+						}
```

Comment:
when is this ever a map? should it not always be a []interface?

Reply:
also if you use v1, k1, ok1, we might want to use sub_value, sub_key. Ok could be assigned to the same ok variable I think

Reply:
To answer the first question it is indeed a map when "resource_required" is used directly under the class name as shown below in the properties.yml as opposed to when used under  test_values ->
```
fvRemoteId:
  overwrites:
    siteId: "site_id"
    remote_ctx_pc_tag: "remote_vrf_pc_tag"
  documentation:
    siteId: "A number between 0 and 1000 to identify the remote site being associated with the primary site"
  default_values:
    siteId: "100"
    remote_ctx_pc_tag: "any"
    remote_pc_tag: "16386"
  resource_required:
    - "remotePcTag"
  test_values:
    all:
      site_id: "100"
      remote_vrf_pc_tag: "any"
      remote_pc_tag: "16386"
    default:
      site_id: "100"
      remote_vrf_pc_tag: "any"
      remote_pc_tag: "16386"
    resource_required:
      site_id: "100"
   ```
   
   That being said we need to refactor the code. We could look into using a switch statement or something similar